### PR TITLE
Fixes #11612 - Enables use of Ostree Units

### PR DESCRIPTION
**Pull request by parthaa**

### CODE REVIEW

**Comment by parthaa** [comment]
continuing from https://github.com/Katello/katello/pull/5838 .. New branch. So others could contribute.. Will squash once completed...

**Comment by parthaa** [comment]
needs Katello/runcible#170

File: app/models/katello/glue/pulp/repo.rb [hunk 1]
```diff
@@ -477,6 +477,26 @@ def pulp_docker_manifest_ids
         Katello.pulp_server.extensions.repository.docker_manifest_ids(self.pulp_id)
       end
 
+      def pulp_ostree_branch_ids
+        Katello.pulp_server.extensions.repository.pulp_ostree_branch_ids(self.pulp_id)
```
Comment (daviddavis) [inline app/models/katello/glue/pulp/repo.rb]: `ostree_branch_ids`

File: app/models/katello/glue/pulp/repo.rb [hunk 2]
```diff
@@ -477,6 +477,26 @@ def pulp_docker_manifest_ids
...
+          branch = OstreeBranch.where(:uuid => ostree_branch_json[:_id]).first_or_create
+          branch.update_from_json(ostree_branch_json)
+        end
+        OstreeBranch.sync_repository_associations(self, pulp_ostree_branch_ids)
+      end
+
+      def ostree_branches_json
+        ostree_branches = []
+        pulp_ostree_branch_ids.each_slice(SETTINGS[:katello][:pulp][:bulk_load_size]) do |sub_list|
+          ostree_branches.concat(Katello.pulp_server.extensions.ostree_branches.find_all_by_unit_ids(sub_list))
```
Comment (daviddavis) [inline app/models/katello/glue/pulp/repo.rb]: `ostree_branch`

File: app/views/katello/api/v2/repositories/base.json.rabl [hunk 3]
```diff
@@ -13,6 +13,7 @@ end
 
 node :content_counts do |repo|
   {
+    :ostree_branches => repo.ostree_branches.count,
```
Comment (daviddavis) [inline app/views/katello/api/v2/repositories/base.json.rabl]: `ostree_branch`

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/views/product-repositories.html [hunk 4]
```diff
@@ -149,6 +149,15 @@
                   </span>
                 </div>
               </span>
+
+              <span ng-show="repository.content_type == 'ostree'">
+                <div>
+                  <span translate>
+                    {{ repository.content_counts.ostree || 0 }} OSTree Units
```
Comment (daviddavis) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/views/product-repositories.html]: `ostree_branch`

**Comment by parthaa** [comment]
Also fixes
Bug #14089 -Ostree Repos should always be protected
Feature #13586: As a user I want to be able to see ostree branches from
summary added
Feature #13588: As a user I want to be able to "Pull All" ostree
branches from summary added
Bug #13695: OSTree repositories show wrong published at URL added
Feature #13720: As a user I want to see ostree information published in
an environment added
Feature #13953: UI - As a user I want to be able to search rpm-ostree
repos added
Bug #11612: OSTree content unit count not updating on repo details page
after sync added

Also Added Manage Branches for Ostree

Add Ostree Search interface for the UI

**Pull request closed without merging**